File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

string solution(string s, int n) {
    int len = s.size();
    string answer = s;
    if (len > 0) {
        for (int i = 0; i < len; i++) {
            if (answer[i] == ' ') continue;
            int idx;
            if (answer[i] - 'a' < 0) {
                idx = answer[i] - 'A' + n;
                answer[i] = 'A';
            }
            else {
                idx = answer[i] - 'a' + n;
                answer[i] = 'a';
            }
            answer[i] += idx % 26;
        }
    } 
    return answer;
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    answer = ''
    for ch in s:
        if 'a' <= ch and ch <= 'z':
            answer += chr(ord('a') + (ord(ch) - ord('a') + n)%26)
        elif 'A' <= ch and ch <= 'Z':
            answer += chr(ord('A') + (ord(ch) - ord('A') + n)%26)
        else:
            answer += ch
    return answer